src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ExactMethodInstantiationsNode.cs

```diff
@@ -0,0 +1,169 @@
+        {
+            // This node does not trigger generation of other nodes.
+            if (relocsOnly)
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolNode[] { this });
+
+            // Zero out the hashset so that we AV if someone tries to insert after we're done.
+            _exactMethodInstantiationsList = null;
+
+            MemoryStream stream = new MemoryStream();
+            _nativeWriter.Save(stream);
+
+            byte[] streamBytes = stream.ToArray();
+
+            _endSymbol.SetSymbolOffset(streamBytes.Length);
+
+            return new ObjectData(streamBytes, Array.Empty<Relocation>(), 1, new ISymbolNode[] { this, _endSymbol });
+        }
+
+        public DependencyList GetExactMethodInstantiationDependenciesForMethod(NodeFactory factory, MethodDesc method)
+        {
+            if (!IsMethodEligibleForTracking(method))
+                return null;
+
+            DependencyList dependencies = new DependencyList();
+
+            dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(method.OwningType), "Exact method instantiation signature"));
+
+            foreach (var arg in method.Instantiation)
+                dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(arg), "Exact method instantiation signature"));
+
+            return dependencies;
+        }
+
+        public void AddExactMethodInstantiationEntry(NodeFactory factory, MethodDesc method)
+        {
+            // Check if we already saw this method
+            if (!_exactMethodInstantiationsList.Add(method))
+                return;
+
+            if (!IsMethodEligibleForTracking(method))
+                return;
+
+            bool getUnboxingStub = (method.OwningType.IsValueType || method.OwningType.IsEnum) && !method.Signature.IsStatic;
+            IMethodNode methodEntryPointNode = factory.MethodEntrypoint(method, getUnboxingStub);
+
+            Vertex methodSignature = GetNativeLayoutInfoSignatureForMethod(factory, method);
+            Vertex methodPointer = _nativeWriter.GetUnsignedConstant(_externalReferences.GetIndex(methodEntryPointNode));
+
+            // Make the generic method entry vertex
+
+            Vertex entry = _nativeWriter.GetTuple(methodSignature, methodPointer);
+
+            // Add to the hash table, hashed by the containing type's hashcode
+            uint hashCode = (uint)method.OwningType.GetHashCode();
+            _hashtable.Append(hashCode, _nativeSection.Place(entry));
+        }
+
+        private bool IsMethodEligibleForTracking(MethodDesc method)
+        {
+            if (method.IsAbstract)
+                return false;
+
+            if (!method.HasInstantiation)
+                return false;
+
+            // Nothing to add for interface methods because they have no implementations of their own...
+            if (method.OwningType.IsInterface)
```

Comment:
> which was not super interesting to cover

If we're discussing adding 3 lines of code whose only purpose is to add a latent bug (to be discovered when the C# team caves and adds support for static methods on interfaces), I would vote to delete the lines. We already had a situation where a C# feature had to be cut because of a bug in the runtime.